pkg/operator/controllers/subnets/subnet_nsg.go

```diff
@@ -30,14 +30,16 @@ func (r *reconcileManager) ensureSubnetNSG(ctx context.Context, s subnet.Subnet)
 		return err
 	}
 
-	if !strings.EqualFold(*subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID) {
-		r.log.Infof("Fixing NSG from %s to %s", *subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID)
+	if subnetObject.SubnetPropertiesFormat.NetworkSecurityGroup != nil && strings.EqualFold(*subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID) {
+		return nil
+	}
 
-		subnetObject.NetworkSecurityGroup = &mgmtnetwork.SecurityGroup{ID: &correctNSGResourceID}
-		err = r.subnets.CreateOrUpdate(ctx, s.ResourceID, subnetObject)
-		if err != nil {
-			return err
-		}
+	oldNSG := "nil"
```

Comment:
might be good to comment here, that because of the previous "if"s, the only two possibilities are either a nil previous nsg or not the right nsg, and that both require reconcilation